### PR TITLE
Deprecate `Operator.expand` function

### PR DESCRIPTION
**Pull request by Shiro-Raven**
Deprecated since it is redundant bloat and can be done manually pretty easily.

[[sc-65069](https://app.shortcut.com/xanaduai/story/65069)]

### CODE REVIEW

File: pennylane/operation.py [hunk 1]
```diff
@@ -1499,9 +1499,19 @@ def adjoint(self) -> "Operator":  # pylint:disable=no-self-use
     def expand(self):
         """Returns a tape that contains the decomposition of the operator.
 
+        .. warning::
+            This function is deprecated and will removed in version 0.39.
```
Comment (trbromley) [inline pennylane/operation.py:1503]:
```suggestion
            This function is deprecated and will be removed in version 0.39.
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -70,6 +70,10 @@
 
 <h3>Deprecations 👋</h3>
 
+* `Operator.expand` has been deprecated. Users should simply do `qml.tape.QuantumScript(op.decomposition())`
+  for an equivalent behaviour.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:74]:
```suggestion
* `Operator.expand` has been deprecated. Users should simply use `qml.tape.QuantumScript(op.decomposition())`
  for equivalent behaviour.
```

File: tests/test_operation.py [hunk 3]
```diff
@@ -321,6 +321,21 @@ class DummyOp(Operator):
         assert op._ndim_params == (ndim_params,)
         assert op.ndim_params == (0,)
 
+    def test_expand_fn_deprecated(self):
```
Comment (mudit2812) [inline tests/test_operation.py:324]: Can you also add a docstring for the test

```suggestion
    def test_expand_deprecated(self):
```

**Review by mudit2812 — APPROVED** [review]

**Review by PietropaoloFrisoni — APPROVED** [review]
:+1:

**Pull request merged**